addons/account/models/account_payment.py

```diff
@@ -940,13 +940,16 @@ def _compute_display_name(self):
             payment.display_name = payment.move_id.name if payment.move_id.name != '/' else _('Draft Payment')
 
     def copy(self, default=None):
-        if not self.is_internal_transfer:
-            default = {
-                'journal_id': self.journal_id.id,
-                'payment_method_line_id': self.payment_method_line_id.id,
-                **(default or {}),
-            }
-        return super().copy(default=default)
+        copied_payments = self.env['account.payment']
+        for payment_record in self:
+            if not payment_record.is_internal_transfer:
+                default = {
+                    'journal_id': payment_record.journal_id.id,
+                    'payment_method_line_id': payment_record.payment_method_line_id.id,
+                    **(default or {}),
+                }
+            copied_payments += super(AccountPayment, payment_record).copy(default=default)
+        return copied_payments
```

Comment:
```suggestion
        copied_payments = self.env['account.payment']
        internal_transfers = self.filtered('is_internal_transfer')
        others = self - internal_transfers
        if internal_transfers:
            copied_payments +=  super(AccountPayment, others).copy(default=default)
        if others:
            default = {
                'journal_id': payment_record.journal_id.id,
                'payment_method_line_id': payment_record.payment_method_line_id.id,
                **(default or {}),
            }
            copied_payments += super(AccountPayment, others).copy(default=default)
        return copied_payments
```

Reply:
seems good, since this version fix the issue of the current version, where we should have rather done:

```diff
+  payment_default = default
   if not payment_record.is_internal_transfer:
-     default = {
+     payment_default = {
           'journal_id': payment_record.journal_id.id,
	   'payment_method_line_id': payment_record.payment_method_line_id.id,
	    **(default or {}),
     }    
-    copied_payments += super(AccountPayment, payment_record).copy(default=default})
+    copied_payments += super(AccountPayment, payment_record).copy(default=payment_default)
```
and batch the calls, but there is the small side effect that the order of copied_payments will not be conserved.